Detectors/TOF/calibration/src/TOFDCSProcessor.cxx

```diff
@@ -0,0 +1,555 @@
+  }
+  if (!mStartTFset) {
+    mStartTF = mTF;
+  }
+
+  std::unordered_map<DPID, DPVAL> mapin;
+  for (auto& it : dps) {
+    mapin[it.id] = it.data;
+  }
+  for (auto& it : mPids) {
+    const auto& el = mapin.find(it.first);
+    if (el == mapin.end()) {
+      LOG(DEBUG) << "DP " << it.first << " not found in map";
+    } else {
+      LOG(DEBUG) << "DP " << it.first << " found in map";
+    }
+  }
+
+  mUpdateFeacStatus = false; // by default, we do not foresee a new entry in the CCDB for the FEAC
+  mUpdateHVStatus = false;   // by default, we do not foresee a new entry in the CCDB for the HV
+
+  // now we process all DPs, one by one
+  for (const auto& it : dps) {
+    // we process only the DPs defined in the configuration
+    const auto& el = mPids.find(it.id);
+    if (el == mPids.end()) {
+      LOG(INFO) << "DP " << it.id << " not found in TOFDCSProcessor, we will not process it";
+      continue;
+    }
+    processDP(it);
+    mPids[it.id] = true;
+  }
+
+  if (mUpdateFeacStatus) {
+    updateFEACCCDB();
+  }
+
+  if (mUpdateHVStatus) {
+    updateHVCCDB();
+  }
+
+  return 0;
+}
+
+//__________________________________________________________________
+
+int TOFDCSProcessor::processDP(const DPCOM& dpcom)
+{
+
+  // processing single DP
+
+  auto& dpid = dpcom.id;
+  const auto& type = dpid.get_type();
+  auto& val = dpcom.data;
+  if (mVerbose) {
+    if (type == RAW_DOUBLE) {
+      LOG(INFO);
+      LOG(INFO) << "Processing DP = " << dpcom << ", with value = " << o2::dcs::getValue<double>(dpcom);
+    } else if (type == RAW_INT) {
+      LOG(INFO);
+      LOG(INFO) << "Processing DP = " << dpcom << ", with value = " << o2::dcs::getValue<int32_t>(dpcom);
+    }
+  }
+  auto flags = val.get_flags();
+  if (processFlags(flags, dpid.get_alias()) == 0) {
+    // now I need to access the correct element
+    if (type == RAW_DOUBLE) {
+      // for these DPs, we will store the first, last, mid value, plus the value where the maximum variation occurred
+      LOG(DEBUG) << "mDpsdoublesmap[dpid].size() = " << mDpsdoublesmap[dpid].size();
+      auto etime = val.get_epoch_time();
+      if (mDpsdoublesmap[dpid].size() == 0 ||
+          etime != mDpsdoublesmap[dpid].back().get_epoch_time()) { // we check
+                                                                   // that we did not get the
+                                                                   // same timestamp as the
+                                                                   // latest one
+        mDpsdoublesmap[dpid].push_back(val);
+      }
+    }
+
+    if (type == RAW_INT) {
+      // for these DPs, we need some processing
+      if (std::strstr(dpid.get_alias(), "FEACSTATUS") != nullptr) { // DP is FEACSTATUS
+        std::string aliasStr(dpid.get_alias());
+        // extracting DDL number (regex is quite slow, using this way)
+        std::size_t const nn = aliasStr.find_first_of("0123456789");
```

Comment:
Actually, I should do either:
```
	const auto offs = std::strlen("TOF_FEACSTATUS");
```
or 
``` 
const auto offs = sizeof(""TOF_FEACSTATUS") - 1;
```
because it is null terminated. I used strlen, is it ok?

Reply:
ok, though you have an "_" after ``TOF_FEACSTATUS``, so like that you will start from it.

Reply:
right, my typo. fixed.